### DISCUSSION TITLE
How can I set `AreVerticalSnapPointsRegular` on a ListBox's ItemsPresenter?

### DISCUSSION THREAD

**Discussion by marklam** [AvaloniaUI/Avalonia#11054] [Q&A] [answered]
(Avalonia v11 Preview 6, themed with Material.Avalonia nightly build)

I'm building a ListBox in code (F# using Avalonia.FuncUI, but I should be able to use a C# answer). The list box has 1M entries, and performance seems fine with a VirtualizingStackPanel - provided I set `AreVerticalSnapPointsRegular` to true on the ItemsPresenter (tested via the Avalonia.Diagnostics property window). If I don't set that, then the snap point logic goes off for ages to query all the items in the list

The Virtualizing StackPanel is set via a FuncTemplate

```F#
ListBox.itemsPanel (FuncTemplate<Panel>(fun () -> VirtualizingStackPanel()))
```

But setting `AreVerticalSnapPointsRegular` on the above VirtualizingStackPanel just gets reset to false - it seems to need to be on the ItemsPresenter (or it only works when set from the debug ui).

Is there a way to set this property when building the UI from code, eg from a Style selector or template (in code) while building the ListBox?

Thanks for any help!

**Comment by maxkatz6** [comment, marked as answer]
ListBox has AreVerticalSnapPointsRegular CLR property too. Should be accessible from F# as well. You don't need to change anything on the VirtualizingStackPanel directly.
Setting this property on ListBox will redirect that value to the ItemsPresenter internally https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Controls/ListBox.xaml#L42

**Comment by marklam** [comment]
Thanks - that was careless of me to miss that, I think I was walking up the visual tree and stopped at the first thing that worked.

Unfortunately I still have the problem that the theme I'd chosen (Material.Avalonia) doesn't have the bindings in the templates to pass that property down. I'll raise an issue there.
